### PR TITLE
TBE-120 Clean up calculate age methods for filers and dependents

### PR DESCRIPTION
**Pull request by mpidcock**
## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/TBE-120

## Is PM acceptance required? (delete one)
- No - merge after code review approval

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- I made the existing `#calculate_age` method in `state_file_base_intake` the base place we do age/birthday calculation (other methods are defined elsewhere that call it)
- Made sure any existing age logic was using this age method, including `senior?`, `under_17?` and sorting dependents by age
- Made some readability updates
- I attempted to fill out all test cases that are MFJ to also have full spouse data, but found that to be a challenge for some tests
- reworked the tests for `state_file_dependent_spec` to focus on `senior?` and `under_17?`

## How to test?
- Regression tests should cover any impacted logic
- Edge cases for Jan 1 are covered in `state_file_base_intake_spec` and `state_file_dependent_spec`

## Screenshots (for visual changes)
Pure refactor, no impact on UI

### CODE REVIEW

File: app/models/state_file_base_intake.rb [hunk 1]
```diff
@@ -372,20 +372,22 @@ def sanitize_bank_details
...
-    calculate_age(inclusive_of_jan_1: true, dob: spouse_birth_date) >= 65
+    calculate_age(spouse_birth_date, inclusive_of_jan_1: true) >= 65
   end
 
-  def calculate_age(inclusive_of_jan_1: true, dob: primary_birth_date)
-    # federal guidelines: you qualify for age related benefits the day before your birthday
-    # that means for a given tax year those born on Jan 1st the following tax-year will be included
-    # this does not apply for benefits you age out of or any age calculations for Maryland
-    raise StandardError, "Primary or spouse missing date-of-birth" if dob.nil?
+  def calculate_age(dob, inclusive_of_jan_1:)
```
Comment (mpidcock) [inline app/models/state_file_base_intake.rb:384]: I changed the order of arguments because the dob is more relevant to the method.

I also removed the default values because I think it is better to always provide them explicitly.

File: app/models/state_file_base_intake.rb [hunk 2]
```diff
@@ -372,20 +372,23 @@ def sanitize_bank_details
...
+    # In tax returns, all ages are calculated based on the last day of the current tax year
+    # Federal exception: for age related benefits, the day before your birthday is when you become older
+    # - Those born on Jan 1st become older on Dec 31st (so are a year older than their birth year would indicate)
+    # - This does not apply for benefits you age out of, such as turning 17 and not being a dependent anymore
+    # - Maryland does not follow the "older on the day before your birthday" rule in any circumstance
```
Comment (anisharamnani) [inline app/models/state_file_base_intake.rb:390]: I appreciate the note here considering that this is used across different state intakes.
